introduction.tex

```diff
@@ -20,22 +20,71 @@ \chapter{Introduction}
 
 \section{Terminology}
 
-A \emph{platform} is a single integrated circuit consisting of one or more
-\emph{components}. Some components may be RISC-V cores, while others may have a different
-function. Typically they will all be connected to a single system bus.
-A single RISC-V core contains one or more hardware threads, called
-\emph{harts}.
-
-\emph{DXLEN} of a hart is its widest supported XLEN, ignoring the current value
-of \Fmxl in \Rmisa.
-
-A \emph{physical address} is directly usable on the system bus.
-
-A \emph{virtual address} is an address as a hart sees it. If there is address
-translation this may be different from the \emph{physical address}. If there is
-no translation then it will be the same.
-
-\subsection{Context}
+\begin{description}[style=nextline]
+    \item[AMO]
+        Atomic Memory Operation.
+    \item[BIST]
+        Built-In Self Test.
+    \item[BYPASS]
+        JTAG instruction that selects a single bit data register, also called BYPASS.
+    \item[component]
+        A RISC-V core, or other part of a platform.
+        Typically all components will all be connected to a single system
+        bus.
+    \item[CSR]
+        Control Status Register.
+    \item[DM]
+        Debug Module.
+    \item[DMI]
+        Debug Module Interface.
+    \item[DR]
+        Data Register.
+    \item[DTM]
+        Debug Transport Module.
+    \item[DXLEN]
+        Debug XLEN, which is the widest XLEN a hart supports, ignoring the
+        current value of \Fmxl in \Rmisa.
+    \item[GPR]
+        General Purpose Register.
+    \item[hart]
+        A hardware thread in a RISC-V core.
+    \item[IDCODE]
+        32-bit Identification CODE, and a JTAG instruction that returns the IDCODE value.
+    \item[IR]
+        Instruction Register.
+    \item[JTAG]
+        Joint Test Action Group.
+    \item[MIE]
+        Machine Interrupt Enable register; also a bit in mstatus.
+    \item[NAPOT]
+        Naturally Aligned Powers-Of-Two.
+    \item[NMI]
+        Non-Maskable Interrupt.
+    \item[OpenOCD]
+        Open On-Chip Debugger.
+    \item[PC]
+        Program Counter.
+    \item[platform]
+        A single integrated circuit consisting of one or more \emph{components}.
```

Comment:
Is this a "platform" or a "hardware platform"?

Reply:
Looking it up, this is a "hardware platform." But do we use that term anywhere?

Reply:
Based on this, I think I need to look at each use of "platform" and probably change it to "hardware platform." Do you agree?

Reply:
I have always found it annoying and confusing that RISC-V uses the word "platform" both for a particular implementation and for standards such as the Unix platform.  But we're stuck with that so "hardware platform" seems appropriate.

Now that I'm looking, I notice that this spec uses both "platform" and "system" to describe the overall implementation:
- "ndmreset resets all the harts in the system"
- "Harts are nonexistent if they will never be part of this system"
- "In most platforms there will only be one DM"
- "There may be multiple DTMs in a single platform."

Is there a distinction?  Maybe "system" (like in the title of chapter 2) means the group of components involved in debug but not all of the components in the whole chip/board/whatever?  Or maybe there's really no distinction?  The base architecture also uses the word "system" so maybe I'm overthinking this.

Reply:
I don't think there's any intentional use of platform vs. system. They should probably all be "hardware platform."